tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -715,6 +716,60 @@ def test_cumprod_numpy_array(self, dtype):
             return cupy.cumprod(a_numpy)
 
 
+@testing.parameterize(*testing.product({
+    'shape': [(20,), (7, 6), (3, 4, 5)],
+    'axis': [None, 0, 1, 2],
+    'func': ('nancumsum', 'nancumprod'),
+}))
+@testing.gpu
+class TestNanCumSumProd(unittest.TestCase):
+
+    density = 0.7
+
+    def _make_array(self, dtype):
+        dtype = numpy.dtype(dtype)
+        if dtype.char in 'efdFD':
+            r_dtype = dtype.char.lower()
+            a = testing.shaped_random(self.shape, numpy, dtype=r_dtype,
+                                      scale=1)
+            if dtype.char in 'FD':
+                ai = a
+                aj = testing.shaped_random(self.shape, numpy, dtype=r_dtype,
+                                           scale=1)
+                ai[ai > math.sqrt(self.density)] = 0
+                aj[aj > math.sqrt(self.density)] = 0
+                a = ai + 1j * aj
+            else:
+                a[a > self.density] = 0
+            a = a / a
+        else:
+            a = testing.shaped_random(self.shape, numpy, dtype=dtype)
+        return a
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_nancumsumprod(self, xp, dtype):
+        if self.axis is not None and self.axis >= len(self.shape):
+            pytest.skip()
+        a = self._make_array(dtype)
+        out = getattr(xp, self.func)(xp.array(a), axis=self.axis)
+        return xp.ascontiguousarray(out)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_nancumsumprod_out(self, xp, dtype):
+        dtype = numpy.dtype(dtype)
+        if self.axis is not None and self.axis >= len(self.shape):
+            pytest.skip()
+        if len(self.shape) > 1 and dtype.char in 'd':
+            # Skip the cases as np.nancum{sum|prod} raises AssertionError.
+            pytest.skip()
```

Comment:
Would you return a dummy array to skip here otherwise the rest of parameterized dtypes are also skipped?
```suggestion
            return xp.array([])  # skip
```

Reply:
Thank you, I did not know that rest of dtypes are to be skipped..

Reply:
It seems we need to skip more cases..

Reply:
np.nancum{sum|prod} seems to raise an exception if `a.ndim > 1` and `axis is None`.  I will skip these cases.

Reply:
FYI, I make `pytest.skip` as we expect for the parameterized dtype test in #4094.

Reply:
Would life be easier if you simply raise `unittest.SkipTest`?

Reply:
It's also fine, but developers need to know they have to raise `unittest.SkipTest` to skip a dtype. How do you think?

Reply:
OK, I will update PR and use `unittest.SkipTest`.

Reply:
> It's also fine, but developers need to know they have to raise `unittest.SkipTest` to skip a dtype. How do you think?

Sorry to reply late, @takagi. IMHO I think it's OK. Personally I don't like `pytest.skip()` because the test suite won't work with `unittest`: https://github.com/pytest-dev/pytest/issues/5832, in which the supposedly skipped tests would be mistakenly marked as failed. But it's just my opinion, and I almost exclusively use `pytest` nowadays.

Reply:
Thanks, I got your point!